realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -60,6 +60,20 @@ RealSenseNodeFactory::~RealSenseNodeFactory()
 	{
 		_query_thread.join();
 	}
+
+	try
+	{
+		_realSenseNode.reset();
+		if (_device)
+		{
+			_device.hardware_reset();
+			_device = rs2::device();
+		}
+	}
+	catch (const rs2::error& e)
+	{
+		ROS_ERROR_STREAM("Exception: " << e.what());
+	}
```

Comment:
I think this hardware reset at the destructor is unnecessary. There is the initial_reset option if needed.

Reply:
Ok, yeah, that makes sense.

Reply:
Hi @malban, Do you think it will be possible for you to remove that part so it could be merged?

Reply:
Yeah, I'll look take care of it today.  Thanks for the reminder.